marine_stress.c

```diff
@@ -0,0 +1,110 @@
+#include "marine.h"
+#include "marine_dev.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+void fill_random(char *buf, size_t len) {
+    for (size_t i = 0; i < len / sizeof(int); ++i) {
+        *(((int *) buf) + i) = rand();
+    }
+}
+
+void random_eth(char *buf) {
+    fill_random(buf, 12); // randomize macs
+    buf[12] = 0x08;
+    buf[13] = 0;
+}
+
+void random_ip(char *buf) {
+    buf += 14; // skip eth
+    *(buf++) = 0x45;
+    *(buf++) = 0;
+    *(buf++) = 0;
+    *(buf++) = 0x0e;
+    *(buf++) = 0;
+    *(buf++) = 0;
+    *(buf++) = 0;
+    *(buf++) = 0;
+    *(buf++) = 0x40;
+    *(buf++) = 0x06;
+    *(buf++) = 0xac;
+    *(buf++) = 0x16;
+    fill_random(buf, 8); // randomize ips
+}
+
+void random_tcp(char *buf) {
+    buf += 14; // skip eth
+    buf += 20; // skip ip
+    fill_random(buf, 4); // randomize ports
+    buf += 4;
+    *(buf++) = 0xde;
+    *(buf++) = 0xad;
+    *(buf++) = 0xbe;
+    *(buf++) = 0xef;
+    *(buf++) = 0;
+    *(buf++) = 0;
+    *(buf++) = 0;
+    *(buf++) = 0;
+    *(buf++) = 0x50;
+    *(buf++) = 0x02;
+    *(buf++) = 0xff;
+    *(buf++) = 0xff;
+    *(buf++) = 0;
+    *(buf++) = 0;
+    *(buf++) = 0;
+    *(buf++) = 0;
+}
+
+int report_mem(size_t rss) {
```

Comment:
Maybe just create function that get the rss, print it, and return it, it can save you alot of lines here
like i think this `report current mem` is redundent cause you already call `get_current_rss()` ea time you want to interace with the rss

Reply:
Or maybe make `report_current_mem` retunr the rss